freqtrade/exchange/exchange.py

```diff
@@ -657,7 +657,9 @@ def get_trades_for_order(self, order_id: str, pair: str, since: datetime) -> Lis
     @retrier
     def get_markets(self) -> List[dict]:
         try:
-            return self._api.fetch_markets()
+            markets = self._api.fetch_markets()
+            self.markets.update({m["symbol"]: m for m in markets})
+            return markets
```

Comment:
This introduces a bug when running the bot because it overwrites the `markets` object (the one used internally by ccxt).

It's not copied on creation (so relies on ccxt's internal refresh method - which is good) - but it's the same CCXT uses internally).

Exception seen:
```
2019-03-03 10:40:48,002 - freqtrade.rpc.rpc_manager - DEBUG - Forwarding message to rpc.telegram
2019-03-03 10:40:48,090 - freqtrade - ERROR - Fatal exception!
Traceback (most recent call last):
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/main.py", line 49, in main
    state = freqtrade.worker(old_state=state)
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/freqtradebot.py", line 131, in worker
    min_secs=min_secs)
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/freqtradebot.py", line 143, in _throttle
    result = func(*args, **kwargs)
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/freqtradebot.py", line 184, in _process
    state_changed = self.process_maybe_execute_buy()
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/freqtradebot.py", line 484, in process_maybe_execute_buy
    if self.create_trade():
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/freqtradebot.py", line 354, in create_trade
    return self.execute_buy(_pair, stake_amount)
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/freqtradebot.py", line 453, in execute_buy
    fee = self.exchange.get_fee(symbol=pair, taker_or_maker='maker')
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/exchange/exchange.py", line 53, in wrapper
    return f(*args, **kwargs)
  File "/home/xmatt/development/python/cryptos/freqtrade/freqtrade/exchange/exchange.py", line 678, in get_fee
    price=price, takerOrMaker=taker_or_maker)['rate']
  File "/home/xmatt/.pyenv/versions/3.7.0/envs/trade3.7/lib/python3.7/site-packages/ccxt/binance.py", line 394, in calculate_fee
    rate = market[takerOrMaker]
KeyError: 'maker'

```

Honestly, I would remove this "updating" method from this PR and work on this separately since it's not directly related to this pr ...